modin/pandas/test/test_io.py

```diff
@@ -1555,6 +1556,19 @@ def test_read_parquet_pandas_index(self, engine, filters):
 
         for col in pandas_df.columns:
             if col.startswith("idx"):
+                # Before this commit, first released in version 2023.1.0, fastparquet relied
+                # on pandas private APIs to handle Categorical indices.
+                # These private APIs broke in pandas 2.
+                # https://github.com/dask/fastparquet/commit/cf60ae0e9a9ca57afc7a8da98d8c0423db1c0c53
```

Comment:
to me this looks like we should actually check for categorical indices in `.read_parquet()` and default-to-pandas there if using `fastparquet` engine, instead of muting the problem in the tests... is this doable?

Reply:
Defaulting to Pandas wouldn't fix the bug -- the same error happens when you use Pandas with (the wrong version of) the fastparquet engine. The only difference as far as I can tell between using Pandas and Modin here is the race condition I mentioned before.

Also, I guess because fastparquet is an optional dependency, we don't actually specify the version range we support anywhere. It's just that the `environment-dev.yml` and `requirements-dev.txt` specified an old version of fastparquet -- so, only contributors to this project (and the CI) were guaranteed to run into this issue. Actual users may have already been using Modin with a newer version of fastparquet, which works with Categorical indices.

Reply:
Oh, right, my bad... thx for bearing with me 😃